### PR TITLE
feat: Add Search Field component

### PR DESCRIPTION
**Pull request by alimpens**
To discuss:

- The 'clear'-button currently uses native functionality, which only works in Chromium browsers (Chrome, Edge), not in Firefox for example. Is this functionality important enough to implement custom functionality for this?

Out of scope:

- Search Field without a button (for instant search result pages, see <https://onderzoek.amsterdam.nl/zoek>). The design is not yet done for that.
- Autosuggest
- Different sizes
- Tests for `SearchFieldButton`, will probably be part of an `IconButton`

### CODE REVIEW

File: packages/react/src/SearchField/SearchFieldInput.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          <VisuallyHidden>{label}</VisuallyHidden>
+        </label>
+        <input
+          {...restProps}
+          autoComplete="off"
+          className={clsx('amsterdam-search-field__input', className)}
+          enterKeyHint="search"
+          id={id}
+          ref={ref}
+          spellCheck="false"
```
Comment (VincentSmedinga) [inline packages/react/src/SearchField/SearchFieldInput.tsx:30]: Shouldn’t we turn this on? It may help people mistyping a word.

**Comment by VincentSmedinga** [comment]
> The 'clear'-button currently uses native functionality, which only works in Chromium browsers (Chrome, Edge), not in Firefox for example. Is this functionality important enough to implement custom functionality for this?

I wouldn’t say so. Firefox users will be accustomed to not having a ‘clear’ button in a search field?

File: packages/react/src/SearchField/SearchFieldInput.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ({ className, label = 'Zoeken', ...restProps }: SearchFieldInputProps, ref: ForwardedRef<HTMLInputElement>) => {
+    const id = useId()
+
+    return (
+      <>
+        <label htmlFor={id}>
+          <VisuallyHidden>{label}</VisuallyHidden>
+        </label>
+        <input
+          {...restProps}
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/react/src/SearchField/SearchFieldInput.tsx:24]: Yes, but I wasn't completely sure about it.  They're both kind of annoying when you're looking for partial text, especially in combination with an autosuggest component. We turned `autoComplete` off at Onderzoek, I'm not sure why we didn't turn off spellcheck. I see data.amsterdam.nl also turns `autoComplete` off.

I thought I stole the `spellCheck` part from gov.uk, but I see they don't have `autoComplete` and `spellCheck` turned off (anymore). 

We could leave it up to the devs?

File: proprietary/tokens/src/components/amsterdam/search-field.tokens.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        "box-shadow": {
+          "value": "inset 0 0 0 1px {amsterdam.color.primary-black}"
+        },
+        "color": { "value": "{amsterdam.color.primary-black}" },
+        "font-family": { "value": "{amsterdam.typography.font-family}" },
+        "font-weight": { "value": "{amsterdam.typography.font-weight.normal}" },
+        "outline-offset": { "value": "{amsterdam.focus.outline-offset}" },
+        "cancel-button": {
+          "background-image": {
+            "value": "url(\"data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32'><path fill='%23004699' fill-rule='evenodd' d='M29.333 5.47 26.53 2.668 16 13.187 5.47 2.666 2.668 5.47 13.187 16 2.666 26.53l2.804 2.803L16 18.813l10.53 10.52 2.803-2.804L18.813 16z'/></svg>\")"
```
[the remark this answers is not in the document]
Comment (alimpens) [reply proprietary/tokens/src/components/amsterdam/search-field.tokens.json:22]: I'm not sure what you mean, do you want to create tokens for all icons?

File: packages/css/src/components/search-field/README.md
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/components/search-field/README.md:1]: Yeah, good idea. Maybe we should wait until we reach a verdict on turning off `autocomplete` and `spellcheck` by default, or leaving it up to the dev?

File: packages/react/src/SearchField/SearchFieldInput.test.tsx
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/react/src/SearchField/SearchFieldInput.test.tsx:1]: No, not really, it's actually a good idea imo. We should test the controlled version of the input

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**